---
uint8 SUCCEEDED=0
uint8 FAILED=1
uint8 result_status
string[] group_names
